packages/cli/generators/relation/belongs-to-relation.generator.js

```diff
@@ -101,10 +111,36 @@ module.exports = class BelongsToRelationGenerator extends BaseRelationGenerator
     await sourceFile.save();
   }
 
-  getBelongsTo(className, relationName, fktype) {
+  getBelongsTo(
+    className,
+    relationName,
+    defaultRelationName,
+    sourceKeyName,
+    fktype,
+  ) {
+    // checks if relation name is customized
+    let relationDecorator = [
+      {
+        name: 'belongsTo',
+        arguments: [`() =>  ${className}`],
+      },
+    ];
+    // except the customized relation name, also need to check if the property name is the same as relation name
+    // or it will get 'navigational property' error when creating instances
```

Comment:
related [stackoverflow question](https://stackoverflow.com/questions/59468329/unhandled-error-in-post-order-products-500-error-navigational-properties-are).